components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -0,0 +1,399 @@
+            return processErrorResponse((ActionInvocationErrorResponse) actionInvocationResponse.getResponse(),
+                    actionType,
+                    eventContext, actionRequest, actionId, apiEndpoint, authenticationMethod,
+                    actionExecutionResponseProcessor);
+        } else {
+            logErrorResponse(actionInvocationResponse, actionType, actionId, apiEndpoint, authenticationMethod);
+        }
+
+        return new ActionExecutionStatus(ActionExecutionStatus.Status.FAILURE, eventContext);
+    }
+
+    private ActionExecutionStatus processSuccessResponse(ActionInvocationSuccessResponse successResponse,
+                                                         ActionType actionType,
+                                                         Map<String, Object> eventContext,
+                                                         ActionExecutionRequest actionRequest,
+                                                         String actionId, String apiEndpoint,
+                                                         AuthMethods.AuthMethod authenticationMethod,
+                                                         ActionExecutionResponseProcessor
+                                                                 actionExecutionResponseProcessor)
+            throws ActionExecutionResponseProcessorException {
+
+        if (LOG.isDebugEnabled()) {
+            // todo: add to diagnostic logs
+            logSuccessResponse(successResponse, actionType, actionId, apiEndpoint, authenticationMethod);
+        }
+
+        List<PerformableOperation> allowedPerformableOperations =
+                validatePerformableOperations(actionRequest, successResponse);
+        ActionInvocationSuccessResponse.Builder successResponseBuilder =
+                new ActionInvocationSuccessResponse.Builder().operations(allowedPerformableOperations);
+        return actionExecutionResponseProcessor.processSuccessResponse(actionType, eventContext,
+                actionRequest.getEvent(),
+                successResponseBuilder.build());
+    }
+
+    private ActionExecutionStatus processErrorResponse(ActionInvocationErrorResponse errorResponse,
+                                                       ActionType actionType,
+                                                       Map<String, Object> eventContext,
+                                                       ActionExecutionRequest actionRequest,
+                                                       String actionId, String apiEndpoint,
+                                                       AuthMethods.AuthMethod authenticationMethod,
+                                                       ActionExecutionResponseProcessor
+                                                               actionExecutionResponseProcessor)
+            throws ActionExecutionResponseProcessorException {
+
+        if (LOG.isDebugEnabled()) {
```

Comment:
Shall we move this `LOG.isDebugEnabled()` check inside the `logErrorResponse()` method. If so can use that method to log the diagnostic logs as well in the future.

Reply:
will move